### PR TITLE
Bugfixes and Tests for /tasklogs/

### PR DESCRIPTION
**Pull request by opalmer**
This change adds full coverage for the /tasklogs/ endpoint and also fixes a couple of latent bugs.

### CODE REVIEW

**Comment by opalmer** [comment]
@guidow, PTAL.  I'm trying to shift my focus a bit to increasing coverage on the agent where we're missing it and this seemed like some low hanging fruit to hit first.  Eventually I want to get back to converting over to inline callbacks....but we need better tests and more coverage first I think.

File: pyfarm/agent/http/api/tasklogs.py [hunk 1]
```diff
@@ -43,23 +87,41 @@ def get(self, **kwargs):
 
         if len(request.postpath) != 1:
             request.setResponseCode(BAD_REQUEST)
+            request.setHeader("Content-Type", "application/json")
```
Comment (guidow) [inline pyfarm/agent/http/api/tasklogs.py]: This is wrong. The content is text/csv.

File: tests/test_agent/test_http_api_tasklogs.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        with open(self.log, "wb") as log:
+            log.write(self.log_content)
+
+    def test_methods_exist_for_schema(self):
+        self.skipTest("No schemas defined")
+
+    def test_master_contacted(self):
+        """
+        Make sure that if the request is coming
+        from the master that is sets the value in
```
Comment (guidow) [inline tests/test_agent/test_http_api_tasklogs.py]: Typo, maybe some rephrasing needed

**Comment by guidow** [comment]
Looks good

**Pull request merged**